openmdao/core/driver.py

```diff
@@ -523,6 +540,52 @@ def get_constraint_metadata(self):
         """
         return self._cons
 
+    def _gather_vars(self, vec, varnames):
+        '''
+        Gathers and returns only variables listed in 
+        `varnames` from the vector `vec`
+        '''
+        local_vars = []
+
+        for key, value in iteritems(vec):
```

Comment:
Should be more efficient to have the outer loop be 'for key in varnames:', shouldn't it?  I'm assuming that varnames would always be a subset of the names in vec.  Is that true?  If so, you would not only have a potentially smaller number of iterations, but also wouldn't have to test for existence of key in vec.